libethereum/ChainParams.cpp

```diff
@@ -51,50 +51,116 @@ ChainParams::ChainParams(string const& _json, h256 const& _stateRoot)
 	*this = loadConfig(_json, _stateRoot);
 }
 
+namespace
+{
+	string const c_sealEngine = "sealEngine";
+	string const c_params = "params";
+	string const c_genesis = "genesis";
+	string const c_accounts = "accounts";
+
+	set<string> const c_knownChainConfigFields =
+		{c_sealEngine, c_params, c_genesis, c_accounts};
+
+	void validateConfig(js::mObject const& _obj)
+	{
+		for (auto const& elm: _obj)
+			if (c_knownChainConfigFields.find(elm.first) == c_knownChainConfigFields.end())
+			{
+				string const comment = "Unknown field in genesis config: " + elm.first;
+				cerr << comment << "\n";
+				BOOST_THROW_EXCEPTION(UnknownField() << errinfo_comment(comment));
+			}
+	}
+
+	string const c_minGasLimit = "minGasLimit";
+	string const c_maxGasLimit = "maxGasLimit";
+	string const c_gasLimitBoundDivisor = "gasLimitBoundDivisor";
+	string const c_homesteadForkBlock = "homesteadForkBlock";
+	string const c_daoHardforkBlock = "daoHardforkBlock";
+	string const c_EIP150ForkBlock = "EIP150ForkBlock";
+	string const c_EIP158ForkBlock = "EIP158ForkBlock";
+	string const c_byzantiumForkBlock = "byzantiumForkBlock";
+	string const c_constantinopleForkBlock = "constantinopleForkBlock";
+	string const c_accountStartNonce = "accountStartNonce";
+	string const c_maximumExtraDataSize = "maximumExtraDataSize";
+	string const c_tieBreakingGas = "tieBreakingGas";
+	string const c_blockReward = "blockReward";
+	string const c_difficultyBoundDivisor = "difficultyBoundDivisor";
+	string const c_minimumDifficulty = "minimumDifficulty";
+	string const c_durationLimit = "durationLimit";
+	string const c_chainID = "chainID";
+	string const c_networkID = "networkID";
+	string const c_allowFutureBlocks = "allowFutureBlocks";
+	string const c_registrar = "registrar";
+
+	set<string> const c_knownParamNames = {
+		c_minGasLimit, c_maxGasLimit, c_gasLimitBoundDivisor, c_homesteadForkBlock,
+		c_EIP150ForkBlock, c_EIP158ForkBlock, c_accountStartNonce, c_maximumExtraDataSize,
+		c_tieBreakingGas, c_blockReward, c_byzantiumForkBlock, c_constantinopleForkBlock,
+		c_daoHardforkBlock, c_minimumDifficulty, c_difficultyBoundDivisor, c_durationLimit,
+		c_chainID, c_networkID, c_allowFutureBlocks, c_registrar
+	};
+
+	void validateParams(js::mObject const& _params)
```

Comment:
Well, this is repetition. How about making one function like `validateFields(jsObject const& _obj, std<string> const& _allowedFields)`?

Reply:
Good idea.